### PR TITLE
Automate apiextensions update

### PR DESCRIPTION
**Pull request by marians**
Closes https://github.com/giantswarm/giantswarm/issues/11124

This adds updating the CRD pages to the workflow that so far updated changes and releases only.

Depends on https://github.com/giantswarm/crd-docs-generator/pull/33

### Please remember to

- [x] Give the PR a meaningful title -- it will be shown in the release notes
- [ ] Add (or remove) [user questions](https://github.com/giantswarm/docs/blob/master/CONTRIBUTING.md#front-matter) associated with any updated docs

### CODE REVIEW

File: scripts/update-crd-reference/main.sh [hunk 1]
```diff
@@ -2,16 +2,22 @@
 
 set -e
 
-CRD_DOCS_GENERATOR_VERSION=0.4.0
+CRD_DOCS_GENERATOR_VERSION=0.4.0-ee74b086d228a80d7c560ffe5a90bf7480c98cd7
```
Comment (marians) [inline scripts/update-crd-reference/main.sh:5]: This depends on https://github.com/giantswarm/crd-docs-generator/pull/33

TODO: After crd-docs-generator is released, update the version here.

File: scripts/update-crd-reference/main.sh [hunk 2]
```diff
@@ -2,16 +2,22 @@
...
 DESTINATION=src/content/ui-api/management-api/crd
 
 # Clear output folder
 find ${DESTINATION} -type f -not -name "_index.md" | xargs rm
 
+# Determine apiextensions version
+VERSION=$(docker run --rm -ti --entrypoint /bin/sh quay.io/giantswarm/docs-scriptrunner -c "curl --silent https://api.github.com/repos/giantswarm/apiextensions/releases/latest|jq -r .tag_name|tr -d '\n'")
+
+echo "Version: ${VERSION}"
```
Comment (ubergesundheit) [inline scripts/update-crd-reference/main.sh:14]: IMO this should be moved into the code of crd-docs-generator.

Comment (marians) [reply scripts/update-crd-reference/main.sh:14]: Is this a blocker for you? If yes, please explain why it shouldn't be done here.

Comment (ubergesundheit) [reply scripts/update-crd-reference/main.sh:14]: Spinning up a container for just a single `curl | jq` call feels a little redundant to me. IMO this belongs into application code. For example the absence of the new flag you've implemented in crd-docs-generator could mean: "Use the latest version of apiextensions". The crd-docs-generator interacts with GitHub after all.

No blocker, but still, feels a little patchwork

Comment (marians) [reply scripts/update-crd-reference/main.sh:14]: OK. If you could approve this and https://github.com/giantswarm/crd-docs-generator/pull/33 I would like to merge, then add the follow-up change.

File: .github/workflows/update-changes.yml [hunk 3]
```diff
@@ -23,20 +25,27 @@ jobs:
...
 
             If you see relevant additions in the 'Files' tab, feel free to **Approve** this PR.
-            (Non-relevant additions would be, for example, a docs release that only updates the Changelog
-            section.)
-            
-            Otherwise, leave the PR open. It will be updated periodically as new entries arrive in the
-            Changes and Releases section.
+
+            If not, leave the PR open. It will be updated periodically as new entries arrive in the
+            Changes and releases section or as CRD pages would get overwritten section.
```
Comment (stone-z) [inline .github/workflows/update-changes.yml:51]:
```suggestion
            Changes and releases section or as CRD page sections are updated.
```

File: scripts/update-crd-reference/main.sh (hunk 1 above)
Comment (marians) [inline scripts/update-crd-reference/main.sh:5]:
```suggestion
CRD_DOCS_GENERATOR_VERSION=0.5.0
```

File: scripts/update-crd-reference/main.sh [hunk 4]
```diff
@@ -2,16 +2,22 @@
...
+
 # Generate new content
 docker run \
     -v ${PWD}/${DESTINATION}:/opt/crd-docs-generator/output \
     -v ${PWD}/scripts/update-crd-reference:/opt/crd-docs-generator/config \
     quay.io/giantswarm/crd-docs-generator:${CRD_DOCS_GENERATOR_VERSION} \
+        --version ${VERSION} \
```
Comment (marians) [inline scripts/update-crd-reference/main.sh:21]:
```suggestion
        --commit-reference ${VERSION} \
```

**Review by ubergesundheit — APPROVED** [review]

**Pull request merged**